### PR TITLE
feat: add bulk insert support for Functions

### PR DESCRIPTION
**Pull request by zhengbuqian**
issue: https://github.com/milvus-io/milvus/issues/35853 and https://github.com/milvus-io/milvus/issues/35856

### CODE REVIEW

File: internal/datanode/importv2/util.go [hunk 1]
```diff
@@ -52,6 +53,7 @@ func NewSyncTask(ctx context.Context,
 	segmentID, partitionID, collectionID int64, vchannel string,
 	insertData *storage.InsertData,
 	deleteData *storage.DeleteData,
+	bm25Stats *map[int64]*storage.BM25Stats,
```
Comment (bigsheeper) [inline internal/datanode/importv2/util.go:56]: map of Go is a reference type, so we can just `bm25Stats map[int64]*storage.BM25Stats`

File: internal/datanode/importv2/util.go [hunk 2]
```diff
@@ -202,6 +207,49 @@ func AppendSystemFieldsData(task *ImportTask, data *storage.InsertData) error {
 	return nil
 }
 
+func RunEmbeddingFunction(task *ImportTask, data *storage.InsertData) error {
```
Comment (bigsheeper) [inline internal/datanode/importv2/util.go:210]: `task Task` is better, and then getting schema by `task.GetSchema()`

File: internal/storage/insert_data.go [hunk 3]
```diff
@@ -69,6 +73,14 @@ func NewInsertDataWithCap(schema *schemapb.CollectionSchema, cap int) (*InsertDa
 		if field.IsPartitionKey && field.GetNullable() {
 			return nil, merr.WrapErrParameterInvalidMsg("partition key field not support nullable")
 		}
+		if field.IsFunctionOutput {
+			if field.IsPrimaryKey || field.IsPartitionKey {
+				return nil, merr.WrapErrParameterInvalidMsg("function output field should not be primary key or partition key")
+			}
+			if !withFunctionOutput {
```
Comment (bigsheeper) [inline internal/storage/insert_data.go:80]: Is it possible to perform an early check?

File: internal/datanode/importv2/util.go (hunk 2 above)
Comment (zhengbuqian) [reply internal/datanode/importv2/util.go:210]: why `task Task` is better? I was following the signature of `AppendSystemFieldsData` method right above this one.

File: internal/storage/insert_data.go [hunk 4]
```diff
@@ -69,6 +73,14 @@ func NewInsertDataWithCap(schema *schemapb.CollectionSchema, cap int) (*InsertDa
 		if field.IsPartitionKey && field.GetNullable() {
 			return nil, merr.WrapErrParameterInvalidMsg("partition key field not support nullable")
 		}
+		if field.IsFunctionOutput {
+			if field.IsPrimaryKey || field.IsPartitionKey {
+				return nil, merr.WrapErrParameterInvalidMsg("function output field should not be primary key or partition key")
```
[the remark this answers is not in the document]
Comment (zhengbuqian) [reply internal/storage/insert_data.go:78]: I don't think it's necessary? the entire schema is logged with every import task:

```
func (t *ImportTask) Execute() []*conc.Future[any] {
	bufferSize := paramtable.Get().DataNodeCfg.ReadBufferSizeInMB.GetAsInt() * 1024 * 1024
	log.Info("start to import", WrapLogFields(t,
		zap.Int("bufferSize", bufferSize),
		zap.Any("schema", t.GetSchema()))...)
```

File: internal/storage/insert_data.go (hunk 3 above)
Comment (zhengbuqian) [reply internal/storage/insert_data.go:80]: can you clarify? what kind of checks are needed?

here we are just making a container object `*InsertData` based on the schema and haven't seen the actual data yet

**Pull request merged**